src/server/bg_services/replication_log_parser.js

```diff
@@ -118,16 +131,18 @@ function aws_parse_log_object(candidates_map, log_object, ignore_fn) {
                 if (ignore_fn(log)) continue;
 
                 if (log.operation.includes('PUT.OBJECT') || log.operation.includes('POST.OBJECT')) {
-                    candidates_map[log.key] = {
+                    candidates.push({
```

Comment:
do you also need to sort actions of the same key by the time in order to make it work correctly?
I wonder if we could do this kind of sorting and then skip redundant actions

Reply:
Sorting by time could be helpful if we can perform more than 1 action. Right now the only truly actionable "action" is deletion. We won't have redundant actions because there can be only 1 successful `DELETE /<key> 204`.

But I think you are right, sorting + the key based approach looks a bit more elegant and makes more sense.

Making this change, thank you! 🙌

Reply:
yes I forgot about the fact that currently we don't support PUT,
for the future it's probably needed

Reply:
I have made the changes. The candidates are sorted by time and if there are conflicting actions (`copy` and `delete` happening at the same time) then the conflict is resolved separately). This would ensure that the sequence of operations that we perform will be correct.